### PR TITLE
update titiler-pgstac version and forward `Request` in reader

### PR DESCRIPTION
**Pull request by vincentsarago**
This PR does:
- update titiler-pgstac/titiler version
- add `Request` in readers attribute and use titiler `reader_dependency` to inject the Request object in the reader at request time

### CODE REVIEW

File: pctiler/setup.py [hunk 1]
```diff
@@ -10,12 +10,12 @@
     "planetary-computer==0.4.*",
 
     "rasterio==1.2.*",
-    "titiler.core==0.5.*",
-    "titiler.mosaic==0.5.*",
+    "titiler.core==0.6.0a2",
+    "titiler.mosaic==0.6.0a2",
```
Comment (vincentsarago) [inline pctiler/setup.py:14]:
```suggestion
    "titiler.core==0.6.0",
    "titiler.mosaic==0.6.0",
```

File: pctiler/pctiler/reader.py [hunk 2]
```diff
@@ -49,6 +67,10 @@ class MosaicSTACReader(pgstac_mosaic.CustomSTACReader):
 
     reader: Type[BaseReader] = attr.ib(default=CustomCOGReader)
 
+    # We make request an optional attribute to avoid re-writing
+    # the whole list of attribute
+    request: Optional[Request] = attr.ib(default=None)
```
Comment (mmcfarland) [inline pctiler/pctiler/reader.py:72]: With `request` now optionally on `MosaicSTACReader`, I expected to be able to access `self.reader.request` within methods of `PGSTACBackend` (like [`assets_for_tiles`](https://github.com/microsoft/planetary-computer-apis/blob/main/pctiler/pctiler/reader.py#L80-L85)), but I get the error 

` AttributeError: type object 'MosaicSTACReader' has no attribute 'request'`

The type annotations suggest this is possible, is that the expected usage? I can confirm that `ReaderParams` was initialized with a request correctly.

Comment (vincentsarago) [reply pctiler/pctiler/reader.py:72]: can you share some code, I have hard to to follow 😓

Comment (mmcfarland) [reply pctiler/pctiler/reader.py:72]: Sure, sorry! One use case for having the request available is to use it for our logging within backend/reader methods, e.g., grabbing a trace id from the header so all logs can be associated with a single request. Usage would be like this: https://github.com/microsoft/planetary-computer-apis/commit/f46100939cb44c6440ec07824ca950e2b35da653, but it doesn't seem like request is available from there.

Comment (vincentsarago) [reply pctiler/pctiler/reader.py:59]: latest commit should add `self.request` (in the Backend)

**Review by mmcfarland — APPROVED** [review]
Works great - I can access the request in both the reader and the backend 👍

**Pull request merged**